pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/AggregationFunctionContext.java

```diff
@@ -54,4 +59,16 @@ public String getColumn() {
   public String getAggregationColumnName() {
     return _aggregationFunction.getColumnName(_column);
   }
+
+  public void setProjectedColumnTypes(final List<FieldSpec.DataType> projectedColumnTypes) {
```

Comment:
Same here, column types should be maintained in the metadata.

Reply:
The source of information is actually coming from metadata as mentioned here https://github.com/apache/incubator-pinot/pull/4535#discussion_r315382449

Just stashing it away in function context to be able to use it during construction of DataTable in InstanceResponeBlock